src/deepsparse/loggers/prometheus_logger.py

```diff
@@ -0,0 +1,116 @@
+# limitations under the License.
+
+"""
+Implementation of the Prometheus Logger
+"""
+import logging
+import os
+from collections import defaultdict
+from typing import Any, Optional
+
+from deepsparse.loggers import BaseLogger, MetricCategories
+from prometheus_client import REGISTRY, Summary, start_http_server, write_to_textfile
+
+
+__all__ = ["PrometheusLogger"]
+
+_LOGGER = logging.getLogger(__name__)
+SUPPORTED_DATA_TYPES = [int, float]
+
+
+class PrometheusLogger(BaseLogger):
+    """
+    DeepSparse logger that continuously exposes the collected logs over the
+    Prometheus python client at the specified port.
+
+    :param port: the port used by the client. Default is 6100
+    :param text_log_save_frequency: the frequency of saving the text log
+        files. E.g. if `text_log_save_frequency` = 10, text logs are dumped
+        after every tenth forward pass. Default set to 10
+    :param text_log_save_dir: the directory where the text log files
+        are saved. By default, the python working directory
+    :param text_log_file_name: the name of the text log file.
+        Default: `prometheus_logs.prom`
+    """
+
+    def __init__(
+        self,
+        port: int = 6100,
+        text_log_save_frequency: int = 10,
+        text_log_save_dir: str = os.getcwd(),
+        text_log_file_name: Optional[str] = None,
+    ):
+
+        self.port = port
+        # until we have done research into persistent
+        # logs, lets save logs as .prom files
+        self.text_log_save_frequency = text_log_save_frequency
+        self.text_log_file_path = os.path.join(
+            text_log_save_dir, text_log_file_name or "prometheus_logs.prom"
+        )
+        self._prometheus_metrics = defaultdict(str)
+
+        self._setup_client()
+        self._counter = 0
+
+    def log(self, identifier: str, value: Any, category: MetricCategories):
+        """
+        Collect information from the pipeline and pipe it them to the stdout
+
+        :param identifier: The name of the thing that is being logged.
+        :param value: The data structure that the logger is logging
+        :param category: The metric category that the log belongs to
+        """
+
+        # needs to adhere to
+        # https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+        formatted_identifier = identifier.replace(".", "__")
+        prometheus_metric = self._prometheus_metrics.get(formatted_identifier)
+        if prometheus_metric is None:
+            prometheus_metric = self._add_metric_to_registry(
+                formatted_identifier, category
+            )
+        prometheus_metric.observe(self._validate(value))
+        self._try_export_metrics_to_textfile()
+
+    def _add_metric_to_registry(self, identifier: str, category: str) -> Summary:
+        description = (
+            """Summary metric for identifier: {identifier} | Category: {category}"""
+        )
+        prometheus_metric = Summary(
```

Comment:
The second small task to take a look at once we get to the end of the engineering work is maybe using so-called labels/exemplars to make logs a bit nicer and structured:
```python
from prometheus_client import Histogram
h = Histogram('request_latency_seconds', 'Description of histogram')
h.observe(4.7, {'trace_id': 'abc123'})
```
(https://github.com/prometheus/client_python)